authorizations/authorization-jdbc/src/main/java/org/apache/gravitino/authorization/jdbc/JdbcAuthorizationPlugin.java

```diff
@@ -0,0 +1,468 @@
+import org.apache.gravitino.authorization.AuthorizationPrivilege;
+import org.apache.gravitino.authorization.AuthorizationSecurableObject;
+import org.apache.gravitino.authorization.Group;
+import org.apache.gravitino.authorization.MetadataObjectChange;
+import org.apache.gravitino.authorization.Owner;
+import org.apache.gravitino.authorization.Role;
+import org.apache.gravitino.authorization.RoleChange;
+import org.apache.gravitino.authorization.SecurableObject;
+import org.apache.gravitino.authorization.User;
+import org.apache.gravitino.connector.authorization.AuthorizationPlugin;
+import org.apache.gravitino.exceptions.AuthorizationPluginException;
+import org.apache.gravitino.meta.AuditInfo;
+import org.apache.gravitino.meta.GroupEntity;
+import org.apache.gravitino.meta.UserEntity;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * JdbcSQLBasedAuthorizationPlugin is the base class for all JDBC-based authorization plugins. For
+ * example, JdbcHiveAuthorizationPlugin is the JDBC-based authorization plugin for Hive. Different
+ * JDBC-based authorization plugins can inherit this class and implement their own SQL statements.
+ */
+@Unstable
+abstract class JdbcAuthorizationPlugin implements AuthorizationPlugin, JdbcAuthorizationSQL {
+
+  private static final String CONFIG_PREFIX = "authorization.jdbc.";
+  public static final String JDBC_DRIVER = CONFIG_PREFIX + "driver";
+  public static final String JDBC_URL = CONFIG_PREFIX + "url";
+  public static final String JDBC_USERNAME = CONFIG_PREFIX + "username";
+  public static final String JDBC_PASSWORD = CONFIG_PREFIX + "password";
+
+  private static final String GROUP_PREFIX = "GRAVITINO_GROUP_";
+  private static final Logger LOG = LoggerFactory.getLogger(JdbcAuthorizationPlugin.class);
+
+  protected BasicDataSource dataSource;
+  protected JdbcSecurableObjectMappingProvider mappingProvider;
+
+  public JdbcAuthorizationPlugin(Map<String, String> config) {
+    // Initialize the data source
+    dataSource = new BasicDataSource();
+    String jdbcUrl = config.get(JDBC_URL);
+    dataSource.setUrl(jdbcUrl);
+    dataSource.setDriverClassName(config.get(JDBC_DRIVER));
+    dataSource.setUsername(config.get(JDBC_USERNAME));
+    dataSource.setPassword(config.get(JDBC_PASSWORD));
+    dataSource.setDefaultAutoCommit(true);
+    dataSource.setMaxTotal(20);
+    dataSource.setMaxIdle(5);
+    dataSource.setMinIdle(0);
+    dataSource.setLogAbandoned(true);
+    dataSource.setRemoveAbandonedOnBorrow(true);
+    dataSource.setTestOnBorrow(BaseObjectPoolConfig.DEFAULT_TEST_ON_BORROW);
+    dataSource.setTestWhileIdle(BaseObjectPoolConfig.DEFAULT_TEST_WHILE_IDLE);
+    dataSource.setNumTestsPerEvictionRun(BaseObjectPoolConfig.DEFAULT_NUM_TESTS_PER_EVICTION_RUN);
+    dataSource.setTestOnReturn(BaseObjectPoolConfig.DEFAULT_TEST_ON_RETURN);
+    dataSource.setLifo(BaseObjectPoolConfig.DEFAULT_LIFO);
+    mappingProvider = new JdbcSecurableObjectMappingProvider();
+  }
```

Comment:
I think we need to provide `JdbcAuthorizationProperties.java` and provide `validate()` function to validate properties.
just like this: https://github.com/apache/gravitino/blob/main/authorizations/authorization-ranger/src/main/java/org/apache/gravitino/authorization/ranger/RangerAuthorizationProperties.java

Reply:
Added.